### DISCUSSION TITLE
PaddleOCR量化模型转onnx报错

### DISCUSSION THREAD

**Discussion by Simyo130** [PaddlePaddle/PaddleOCR#13982] [Q&A]
1、使用官网提供的量化模型，如下所示：
![image](https://github.com/user-attachments/assets/f90a4a24-a56c-49f1-a4bf-d7ab802c6f1b)

**问题描述**
使用下述指令转换：
`paddle2onnx --model_dir . --model_filename inference.pdmodel --params_filename inference.pdiparams --deploy_backend onnxruntime --save_file ./rec_slim.onnx --opset_version 11 --enable_onnx_checker True`

 - Paddle2ONNX版本 : 1.2.9

**报错截图**

![image](https://github.com/user-attachments/assets/ec6beb47-32be-419e-8a20-33bb86471ccb)

2、若使用官网非量化模型，如下所示：
![image](https://github.com/user-attachments/assets/8b8db476-e32d-41d8-9a9d-55e4fbed3be1)

通过动态离线量化方式手动量化上述模型：
```python
import paddle
import paddleslim

paddle.enable_static()
model_dir = "./ch_PP-OCRv3_rec_infer"
model_filename = 'inference.pdmodel'
params_filename = 'inference.pdiparams'
model_dir_quant_dynamic = "./output_rec"

paddleslim.quant.quant_post_dynamic(
    model_dir=model_dir, # 输入模型路径
    model_filename=model_filename, 
    params_filename=params_filename,
    save_model_dir=model_dir_quant_dynamic, 
    save_model_filename=model_filename, 
    save_params_filename=params_filename,
    weight_bits=8, 
)
```
再通过paddle2onnx进行模型转换：
```
paddle2onnx --model_dir . --model_filename inference.pdmodel --params_filename inference.pdiparams --deploy_backend onnxruntime --save_file ./rec_slim.onnx --opset_version 11 --enable_onnx_checker True
``` 

可以成功转换onnx模型，但在使用onnxruntime推理时，报如下错误：
```
 sess = C.InferenceSession(session_options, self._model_path, True, self._read_config_from_model)
onnxruntime.capi.onnxruntime_pybind11_state.InvalidGraph: [ONNXRuntimeError] : 10 : INVALID_GRAPH : Load model from ./onnxocr/models/ppocrv4/rec/model.onnx failed:This is an invalid model. Type Error: Type 'tensor(int8)' of input parameter (conv2d_37.w_0) of operator (Conv) in node (p2o.Conv.0) is invalid.
``` 

请问我该如何解决得到onnx的量化模型？

**Comment by SunYabin0329** [comment]
PaddlOCR量化转ONNX这块问题比较多，官方文档也比价混乱，
我也在尝试对模型进行量化并转为onnx进行推理，感觉量化后推理速度没什么变化

**Reply by SunYabin0329** [reply]
不知道是不是版本问题，官方说的三个文件中，scale文件从来都没见过。
![2024-10-17_17-13-26](https://github.com/user-attachments/assets/3bdd5acf-bf51-4bb1-a80b-9dcd9f9b36f6)
